### PR TITLE
W-13171001-HL7-MLLP-Ref-Updates-HL7 MLLP Connector - Reference Updates

### PR DESCRIPTION
**Pull request by dhirsch1**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 1]
```diff
@@ -4,385 +4,470 @@
...
+| Linger a| Number |  SO_LINGER value, in milliseconds, which indicates how long the socket takes to close so that remaining data is transmitted correctly. |  |
+| Keep Alive a| Boolean | If `true`, enables SO_KEEPALIVE behavior on open sockets. This automatically checks socket connections that are open, but unused, for long periods and closes them if the connection becomes unavailable. This is set on the socket. It is used by a server socket to control whether connections to the server are kept alive before they are recycled. |  false |
+| Fail On Unresolved Host a| Boolean |  Whether the socket fails during its creation if the host set on the endpoint can't be resolved. Set this value to `false` to allow unresolved hosts (useful when connecting through a proxy). |  true |
+|Connection Count |Number | Number of connections accepted at one time. This  value can be an integer or DataWeave expression. An empty value, negative value, or value of `0` indicates that the connector can accept an unlimited number of connections. | |
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
|Connection Count |Number | Number of connections accepted at one time. This value can be an integer or DataWeave expression. An empty value, negative value, or value of `0` indicates that the connector can accept an unlimited number of connections. | |
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 2]
```diff
@@ -4,385 +4,470 @@
...
 |===
 
-
-=== Associated Sources
-
-* <<mllp-listener>>
-
----
 [[request-config]]
 == Request Config
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
== Request Configuration
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 3]
```diff
@@ -4,385 +4,470 @@
...
+[[mllp-listener]]
+=== MLLP Listener
 
-`<mllp:send>`
+`<mllp:mllp-listener>`
 
-Sends the data using the client associated to the MllpRequesterConnection.
+Listens for the socket connections of the given protocol in the configured host and port. When a new connection is received, this source schedules a socket worker to handle the communication for that connection.
 
 === Parameters
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion

```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 4]
```diff
@@ -4,385 +4,470 @@
...
+* <<repeatable-in-memory-stream>>
+* <<repeatable-file-store-stream>>
+* non-repeatable-stream |  Configures how Mule processes streams. Repeatable streams are the default behavior. |  |
+| Redelivery Policy a| <<RedeliveryPolicy>> | Configures the redelivery policy for executing requests that generate errors. You can add a redelivery policy to any source in a flow.   |  |
 | Reconnection Strategy a| * <<reconnect>>
-* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
+* <<reconnect-forever>> | Reconnection strategy to use. |  |
 |===
 
+=== Output
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
==== Output
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 5]
```diff
@@ -4,385 +4,470 @@
...
-* MLLP:RETRY_EXHAUSTED
-* MLLP:SENDING_ERROR
-* MLLP:UNKNOWN
+[%autowidth.spread]
+|===
+| Type | Binary
+| Attributes Type a| <<ImmutableSocketAttributes>>
+|===
 
+=== For Configurations
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
=== Associated Configurations
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 6]
```diff
@@ -4,385 +4,470 @@
...
 
+[[send]]
+=== Send
 
-Listens for socket connections of the given protocol in the configured host and port. When a new connection is received, this source schedules a SocketWorker to handle the communication for that particular connection.
+`<mllp:send>`
 
+Sends the data using the client associated with the MLLP Requester connection.
 
 === Parameters
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 7]
```diff
@@ -4,385 +4,470 @@
...
-| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message. |  |
+| Configuration | String | Name of the configuration to use. | |x
+| Content a| Binary |  Data to serialize and send through the socket. |  `#[payload]` |
 | Reconnection Strategy a| * <<reconnect>>
-* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
+* <<reconnect-forever>> |  Reconnection strategy to use. |  |
 |===
 
-=== Output
+=== For Configurations
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
=== Associated Configurations
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 8]
```diff
@@ -4,385 +4,470 @@
...
 
-[%autowidth.spread]
-|===
-| Type | Binary
-| Attributes Type a| <<ImmutableSocketAttributes>>
-|===
+* <<request-config>>
 
-=== For Configurations
+=== Throws
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
==== Throws
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 9]
```diff
@@ -4,385 +4,470 @@
...
-* non-repeatable-stream |  Configure if repeatable streams should be used and their behavior. |  |
-| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message. |  |
+| Configuration | String | Name of the configuration to use. | |x
+| Content a| Binary |  Data to serialize and send through the socket. |  `#[payload]` |
 | Reconnection Strategy a| * <<reconnect>>
-* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
+* <<reconnect-forever>> |  Reconnection strategy to use. |  |
 |===
 
-=== Output
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:159]: Did you mean to delete this "Output" section?

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 10]
```diff
@@ -4,385 +4,470 @@
...
+=== Throws
 
-* <<mllp-listener-config>>
+* MLLP:CONNECTIVITY
+* MLLP:LISTENING_ERROR
+* MLLP:RETRY_EXHAUSTED
+* MLLP:SENDING_ERROR
+* MLLP:UNKNOWN
 
 == Types
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
== Object Types
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 11]
```diff
@@ -4,385 +4,470 @@
...
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:169]: Add bulleted list with links to each object type

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 12]
```diff
@@ -4,385 +4,470 @@
...
+| Host Name a| String | Name of the host where the requests are sent.  |  |
+| Local Certificates a| Array of Any | SSL local certificates. If the socket is not an SSL socket, this value is null.  |  |
+| Peer Certificates a| Array of Any | SSL peer certificates. If the socket is not an SSL socket, this value is null. |  |
+| Port a| Number | Port to which the requests are sent. |  |
 |===
 
-[[custom-ocsp-responder]]
-=== Custom OCSP Responder
+[[KeyStore]]
+=== KeyStore
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
=== Keystore
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 13]
```diff
@@ -4,385 +4,470 @@
...
 [[reconnect-forever]]
 === Reconnect Forever
 
+Configures a forever reconnection strategy by which the connector source or operation attempts to reconnect at a specified frequency for as long as the Mule app runs.
+
 [%header%autowidth.spread]
 |===
 | Field | Type | Description | Default Value | Required
-| Frequency a| Number | How often in milliseconds to reconnect. |  |
+| Frequency a| Number | How often to reconnect, in milleseconds. |  |
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
| Frequency a| Number | How often to reconnect, in milliseconds. |  |
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 14]
```diff
@@ -4,385 +4,470 @@
...
+
+* Configure a larger buffer size to avoid the number of times Mule needs to write the buffer on disk. This increases performance, but it also limits the number of concurrent requests your application can process, because it requires additional memory.
+* Configure a smaller buffer size to decrease memory load at the expense of response time.|  |
 | Buffer Unit a| Enumeration, one of:
 
 ** BYTE
 ** KB
 ** MB
-** GB | The unit in which all these attributes are expressed. |  |
+** GB | Unit for the *In Memory Size* field. |  |
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
** GB | Unit for the *Max In Memory Size* field. |  |
```

Comment (sathya0) [reply hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]: Or is the field called *In Memory Size*, in which case you should change the name in the above line?

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 15]
```diff
@@ -4,385 +4,470 @@
...
+* <<custom-ocsp-responder>>
+* <<crl-file>> | Configures a revocation checking mechanism. |  |
+|===
 
-** WHEN_EXHAUSTED_GROW
-** WHEN_EXHAUSTED_WAIT
-** WHEN_EXHAUSTED_FAIL | Specifies the behavior of the Mule component pool when the pool is exhausted. Possible values are: "WHEN_EXHAUSTED_FAIL", which throws a NoSuchElementException, "WHEN_EXHAUSTED_WAIT", which blocks by invoking Object.wait(long) until a new or idle object is available, or WHEN_EXHAUSTED_GROW, which  creates a new Mule instance and returns it, essentially making maxActive meaningless. If a positive maxWait value is supplied, it blocks for at most that many milliseconds, after which a NoSuchElementException is thrown. If maxThreadWait is a negative value, it blocks indefinitely. |  |
-| Initialization Policy a| Enumeration, one of:
+[[TrustStore]]
+=== TrustStore
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc]:
```suggestion
=== Truststore
```

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 16]
```diff
@@ -4,385 +4,470 @@
 [[mllp-listener-config]]
 == MLLP Listener Configuration
 
-=== Parameters
-
 [%header%autowidth.spread]
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:7]: global: change [%header%autowidth.spread] to [%header,cols="20s,25a,30a,15a,10a"] so the fields appear bold

Comment (dhirsch1) [reply hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:7]: I don't agree that the fields should be bold in the tables. IMO, it is too much bold and detracts from the docs. We can discuss this at a scrum.

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc (hunk 9 above)
Comment (dhirsch1) [reply hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:159]: Yes, there was nothing in it.

Comment (dhirsch1) [reply hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:159]: It isn't in the original. I am not sure why it shows up here.

File: hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc [hunk 17]
```diff
@@ -4,385 +4,484 @@
...
-* <<mllp-listener-config>>
+* MLLP:CONNECTIVITY
+* MLLP:LISTENING_ERROR
+* MLLP:RETRY_EXHAUSTED
+* MLLP:SENDING_ERROR
+* MLLP:UNKNOWN
 
-== Types
+== Object Types
+
```
Comment (sathya0) [inline hl7-mllp/2.1/modules/ROOT/pages/hl7-mllp-connector-reference.adoc:151]: add Custom OCSP Responder to this list

**Review by sathya0 — APPROVED** [review]

**Pull request merged**